Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -139,7 +139,7 @@ def check_openssl_links
     keg = Keg.new(formula.prefix)
     system_openssl = keg.mach_o_files.select do |obj|
       dlls = obj.dynamically_linked_libraries
-      dlls.any? { |dll| /\/usr\/lib\/lib(crypto|ssl).(\d\.)*dylib/.match dll }
+      dlls.any? { |dll| /\/usr\/lib\/lib(crypto|ssl).(\d\.)*(dylib|so)/.match dll }
```

Comment:
This regexp matches `/usr/lib/libcrypto.dylib` and `/usr/lib/libssl.dylib` (and the versioned variants). Since there won't ever be a `.so` file in `/usr/lib`, I don't think this is changing what you want.